### PR TITLE
add transaction rprompt

### PR DESCRIPTION
**Pull request by guoweikuang**
New redis transaction prompt, then the user will know he is in a transection.
https://github.com/laixintao/iredis/issues/92

### CODE REVIEW

**Comment by guoweikuang** [comment]
#92

File: iredis/entry.py [hunk 1]
```diff
@@ -113,6 +123,12 @@ def gather_args(ctx, h, p, n, password, raw, cmd, decode):
 
     When no command is given, redis-cli starts in interactive mode.
 
+    \b
```
Comment (laixintao) [inline iredis/entry.py]: Excellent!

File: iredis/entry.py [hunk 2]
```diff
@@ -55,11 +57,13 @@ def repl(client, session, start_time):
             _interval = None
 
         try:
+            get_rprompt = lambda: '<transaction>' if in_transaction else None
```
Comment (laixintao) [inline iredis/entry.py]: Do we really need a lambda here?

Comment (laixintao) [reply iredis/entry.py]:
```suggestion
            get_rprompt = '<transaction>' if in_transaction else None
```

Comment (laixintao) [reply iredis/entry.py]: I think this line should be out of try-catch scope. It can't generate any Exceptions.

Comment (guoweikuang) [reply iredis/entry.py]: update

**Comment by laixintao** [comment]
There is a callback mechanism, means that if a command executed successfully, the callback will be executed to render output and do something else.

For example, update the completers, or update the config state.

https://github.com/laixintao/iredis/blob/988833a2081b0d6f967708451ba45616793b56c9/iredis/renders.py#L122

How about we use the global [config](https://github.com/laixintao/iredis/blob/master/iredis/config.py) to flag transaction/un-transaction, and use callback function to update the config?

**Comment by guoweikuang** [comment]
This is a good idea, I will try

**Comment by laixintao** [comment]
We use pexpect to mock terminal test: https://pexpect.readthedocs.io/en/stable/

If it raise Timeout, it means iredis didn't write the output as we expected.

**Comment by guoweikuang** [comment]
@laixintao  May not be caused by the PR I submitted, I am looking for a reason

**Comment by laixintao** [comment]
Thanks! All looks good.

File: iredis/renders.py [hunk 3]
```diff
@@ -113,6 +113,30 @@ def render_ok(text, completer):
...
+    if text is None:
+        return NIL
+    text = _ensure_str(text)
+    config.transaction = False
+    assert text == "OK"
+    return FormattedText([("class:success", text)])
+
+
+def render_exec_ok(items, completer):
+    print('render exec ok')
```
Comment (laixintao) [inline iredis/renders.py]: please delete this print :)

**Comment by laixintao** [comment]
There are some tricky situations here, like when user suffers a connection problem, and client raise an exception, I don't think we can set the transaction to `False`.

Merge for now, chekc later :)

**Pull request merged**